core/src/main/scala/chisel3/Data.scala

```diff
@@ -64,7 +64,7 @@ object SpecifiedDirection {
     val prevId = Builder.idGen.value
     val data = source // evaluate source once (passed by name)
     requireIsChiselType(data)
-    val out = if (!data.mustClone(prevId)) data else data.cloneType.asInstanceOf[T]
+    val out = if (!data.mustClone(prevId)) data else data.cloneTypeFull.asInstanceOf[T]
     out.specifiedDirection = dir(data) // Must use original data, specified direction of clone is cleared
```

Comment:
the direction stripped-ness is overridden just below here. Curious why we wouldn't just follow the same pattern here and override the `out.probeModifier = probeMondifier(data)` ? (apologies for my pseudocode)

Reply:
Does `.cloneTypeFull` also propagate specified direction? Maybe we can delete the line below?

Reply:
> the direction is overridden just below here. Curious why we wouldn't just follow the same pattern here and override the `out.probeModifier = probeModifier(data)` ? (apologies for my pseudocode)

Good question, thank you.  Trying to answer it revealed an important detail I had missed re:how `cloneTypeFull` works, particularly that it does /not/ copy modifiers (const/probe/direction) recursively (!).

As to your suggestion, IMHO it seems best to let `cloneTypeFull` worry about what all a full clone requires (instead of maintaining same logic here), and only change what we need (`specifiedDirection`) after letting it do the full clone for us.  That way any future modifiers or changes to what a full clone mean only need to update `cloneTypeFull` and this code only concerns itself with its task-- setting a new direction.

While not copying modifiers recursively has implications for a use case I am working on (Debug/Probe-blasting),
I believe this change is useful anyway to preserve the sorts of top-level modifications users can do via `ProbeValue` or `Const` so `Output` and related don't drop them (as with the test example)

 I'm not sure if user code can set non-top-level modifiers presently or otherwise run into this behavior in surprising ways, but offhand it seems like at very least that's harder to do... but I don't have a handle on the limits/patterns of user capabilities in Chisel.
(LMK if that sounds right/wrong or any hunches on that front, :+1:)

> Does `.cloneTypeFull` also propagate specified direction? Maybe we can delete the line below?

It does!
However, this code needs to compute the new specified direction using the provided `dir: T => SpecifiedDirection`.